### ISSUE TITLE
Library combine inconsistencies

### ISSUE THREAD

**Issue by c-dante** [paldepind/flyd#73]
While doing the refactor from https://github.com/paldepind/flyd/pull/71 I noticed that both `scan` and `merge` filter out undefined values.

For dependent streams, the combinator returning undefined won't update the stream:
```
var returnVal = s.fn(s, s.depsChanged);
if (returnVal !== undefined) {
  s(returnVal);
}
```
(https://github.com/paldepind/flyd/blob/master/lib/index.js#L63)

Scan and merge (and any dependent streams that return a value instead of using self) will filter undefined.
(https://github.com/paldepind/flyd/blob/master/lib/index.js#L23)

I think internal functions should preserve the `undefined` event and pass it along to merge and scan.

**Comment by paldepind** [comment]
I agree. This is an oversight. I think they should both pass their value on by invoking `self` similar to how `map` does it.

**Comment by c-dante** [comment]
This can be closed -- it was addressed in #71 => https://github.com/paldepind/flyd/commit/32baa28e2c2a4e3cba4746f5b85f9011333536e7
